test/currency-test.js

```diff
@@ -0,0 +1,58 @@
+import { assert } from 'chai';
+import Currency, { CURRENCY_TYPES, CURRENCY_UNITS } from '../lib/currency';
+import { Amount, pkg } from 'hsd';
+
+describe.only('Currency', () => {
+  let bitcoin, bitcoinCash, handshake, baseAmt;
+  beforeEach(() => {
+    baseAmt = 5;
+    bitcoin = new Currency('bitcoin', 5);
+    bitcoinCash = new Currency('bitcoincash', 5);
+    handshake = new Currency('handshake', 5);
+  });
+
+  it('should be an instance of HSD Amount class', () => {
+    assert.instanceOf(handshake, Amount);
+    assert.instanceOf(bitcoin, Amount);
+    assert.instanceOf(bitcoinCash, Amount);
+  });
+
+  it('should return matching units for Handshake', () => {
+    assert.equal(handshake.getUnit(CURRENCY_TYPES.base), pkg.base);
+    assert.equal(handshake.getUnit(CURRENCY_TYPES.micro), `u${pkg.unit}`);
+    assert.equal(handshake.getUnit(CURRENCY_TYPES.milli), `m${pkg.unit}`);
+    assert.equal(handshake.getUnit(CURRENCY_TYPES.unit), pkg.unit);
+    assert.equal(handshake.getUnit(CURRENCY_TYPES.currency), pkg.currency);
+  });
+
+  it('should return correct units for Bitcoin and Bitcoin Cash', () => {
+    const currencies = [bitcoin, bitcoinCash];
+    currencies.forEach(currency => {
+      assert.equal(currency.getUnit(CURRENCY_TYPES.base), 'satoshi');
```

Comment:
It feels like the `pkg` way of doing it should be backported to `bcoin`

Reply:
I totally agree with you 😁 https://github.com/bcoin-org/bcoin/pull/623